File: legds/fun1-1.1/java/Main.java
import java.util.Scanner;

import src.BinSearch;

public class Main{

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);

		System.out.printf("Please enter a length of array%n");
		int length = scan.nextInt();

		System.out.printf("Please enter a ordered array%n");
		int[] arr = new int[length];
		for(int i = 0;i < arr.length;i ++) {
			arr[i] = scan.nextInt();
		}

		System.out.printf("Please enter match%n");
		int match = scan.nextInt();
		scan.close();

		// int re = BinSearch.binSearch(arr, match);
		// if(re >=0) {
		// 	System.out.printf("ordered_array[%d] = match%n",re);
		// } else {
		// 	System.out.printf("Don't find in the ordered_array%n");
		// }
		int re = BinSearch.binSearch(arr,0,arr.length-1,match);
		if(re >=0) {
			System.out.printf("ordered_array[%d] = match%n",re);
		} else {
			System.out.printf("Don't find in the ordered_array%n");
		}
	}	
}

File: legds/fun1-1.1/c/main.c
#include "main.h"

int main() {
	int length;

	printf("Please enter a length\n");
	scanf("%d",&length);

	int arr[length];	
	printf("Please enter a ordered array\n");
	for(int i = 0;i < length;i ++) {
		scanf("%d",&arr[i]);
	}

	int match;
	printf("Please enter match\n");
	scanf("%d",&match);

	int re = BinSearch(arr,length,match);
	if(re >= 0) {
		printf("ordered_array[%d] = match\n",re);
	} else {
		printf("Don't find in the ordered_array\n");
	}
	// int re = BinSearch0(arr,0,length-1,match);
	//
	// if(re >= 0) {
	// 	printf("ordered_array[%d] = match\n",re);
	// } else {
	// 	printf("Don't find in the ordered_array\n");
	// }
	return 0;
}
